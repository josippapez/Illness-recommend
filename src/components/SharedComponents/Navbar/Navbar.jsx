import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink, useHistory } from 'react-router-dom';

import './Navbar.scss';
import { logOutAndWipeLocalStorage } from '../../../interceptor';
import { setDarkTheme } from '../../../store/actions';

import Logo from '../../../styles/assets/images/Logo.svg';
import LogoWithWhiteText from '../../../styles/assets/images/LogoWhiteText.svg';
import HomeImage from '../../../styles/assets/images/home.svg';
import HomeImageWhite from '../../../styles/assets/images/homeWhite.svg';
import UserDetailsImage from '../../../styles/assets/images/resume.svg';
import UserDetailsImageWhite from '../../../styles/assets/images/resumeWhite.svg';
import MedicationSuggestionImage from '../../../styles/assets/images/medicine.svg';
import MedicationSuggestionImageWhite from '../../../styles/assets/images/medicineWhite.svg';
import UsersListImage from '../../../styles/assets/images/user.svg';
import UsersListImageWhite from '../../../styles/assets/images/userWhite.svg';
import LogoutImage from '../../../styles/assets/images/login-arrow-symbol-entering-back-into-a-square.svg';
import LogoutImageWhite from '../../../styles/assets/images/login-arrow-symbol-entering-back-into-a-square-white.svg';

const Navbar = () => {
  const history = useHistory();
  const dispatch = useDispatch();

  const theme = useSelector(state => state.theme);
  const user = useSelector(state => state.user);
  const patient = useSelector(state => state.patient.currentPatientInfo);

  return (
    user.isAuthenticated &&
    window.location.pathname !== '/login' && (
      <div className="navbar">
        <div
          className="navbar-logo"
          onClick={() => {
            history.replace('/');
          }}
        >
          <img
            className="navbar-logo__image"
            src={theme.darkTheme ? LogoWithWhiteText : Logo}
            alt="logo-image"
          />
        </div>
        <div className="darkmode-switch">
          <input
            type="checkbox"
            className="darkmode-switch__input"
            id="Switch"
            readOnly
            checked={!theme.darkTheme}
            onClick={() => dispatch(setDarkTheme(!theme.darkTheme))}
          />
          <label className="darkmode-switch__label" htmlFor="Switch">
            <span className="darkmode-switch__indicator"></span>
            <span className="darkmode-switch__decoration"></span>
          </label>
        </div>
        <div className="navbar-links">
          <NavLink
            className="navbar-item"
            exact
            to="/"
            activeClassName="active"
          >
            <img
              src={theme.darkTheme ? HomeImageWhite : HomeImage}
              alt="home"
              className="navbar-icon"
            />
            Početna stranica
          </NavLink>
          {user.role === 'admin' && (
            <div>
              <NavLink
                className="navbar-item"
                to="/users"
                activeClassName="active"
              >
                <img
                  src={theme.darkTheme ? UsersListImageWhite : UsersListImage}
                  alt="home"
                  className="navbar-icon"
                />
                Liječnici
              </NavLink>
              <NavLink
                className="navbar-item"
                to="/medications"
                activeClassName="active"
              >
                <img
                  src={
                    theme.darkTheme
                      ? MedicationSuggestionImageWhite
                      : MedicationSuggestionImage
                  }
                  alt="home"
                  className="navbar-icon"
                />
                Lijekovi
              </NavLink>
            </div>
          )}
          {user.role === 'user' && (
            <div>
              <NavLink
                className="navbar-item"
                to="/patient-details"
                activeClassName="active"
              >
                <img
                  src={
                    theme.darkTheme ? UserDetailsImageWhite : UserDetailsImage
                  }
                  alt="home"
                  className="navbar-icon"
                />
                Detalji pacijenta
              </NavLink>
              {patient && (
                <NavLink
                  className="navbar-item"
                  to="/medication-suggestion"
                  activeClassName="active"
                >
                  <img
                    src={
                      theme.darkTheme
                        ? MedicationSuggestionImageWhite
                        : MedicationSuggestionImage
                    }
                    alt="home"
                    className="navbar-icon"
                  />
                  Prijedlog lijekova
                </NavLink>
              )}
            </div>
          )}
          <NavLink
            className="navbar-item"
            to="/patients-details-history"
            activeClassName="active"
          >
            <img
              src={theme.darkTheme ? UserDetailsImageWhite : UserDetailsImage}
              alt="home"
              className="navbar-icon"
            />
            {user.role === 'admin' ? 'Podaci pacijenata' : 'Prošli pacijenti'}
          </NavLink>
          <button
            className="navbar-item"
            onClick={() => {
              logOutAndWipeLocalStorage();
            }}
          >
            <img
              src={theme.darkTheme ? LogoutImageWhite : LogoutImage}
              alt="home"
              className="navbar-icon"
            />
            Odjava
          </button>
        </div>
      </div>
    )
  );
};

export default Navbar;
