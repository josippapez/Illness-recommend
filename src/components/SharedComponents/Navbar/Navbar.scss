.navigation-setup {
  display: flex;
}

.navigation-setup:is(.darkTheme) {
  .navbar {
    background-color: var(--dark-background);
    .navbar-links {
      .navbar-item {
        background: var(--dark-navbar-items) 0% 0% no-repeat padding-box;
        color: var(--white);
        &.active {
          background: var(--page-content-background);
        }
        &:hover:not(.active) {
          background: var(--faded-gray) 0% 0% no-repeat padding-box;
          color: var(--main-text);
        }
      }
    }
  }
}

.navbar {
  width: 20%;
  height: 100vh;
  min-width: 300px;
  background-color: var(--unnamed-color-e2e2e3);
  display: flex;
  flex-direction: column;
  position: fixed;
  transition: background-color 0.2s ease-in-out;
  .navbar-logo {
    cursor: pointer;
    padding: 40px 20px 40px 20px;
    height: 20%;
    -webkit-animation: fade-in 0.3s cubic-bezier(0.39, 0.575, 0.565, 1) both;
    animation: fade-in 0.3s cubic-bezier(0.39, 0.575, 0.565, 1) both;
    &__image {
      width: 100%;
    }
  }
  .navbar-links {
    position: relative;
    top: 15%;
    -webkit-animation: fade-in 0.3s cubic-bezier(0.39, 0.575, 0.565, 1) both;
    animation: fade-in 0.3s cubic-bezier(0.39, 0.575, 0.565, 1) both;
    .navbar-icon {
      height: 19px;
      margin-right: 10px;
    }
    .navbar-item {
      margin: 20px 0 20px 10px;
      text-decoration: none;
      display: flex;
      cursor: pointer;
      width: calc(100% - 30px);
      height: 50px;
      border: none;
      border-radius: 50px 10px 10px 50px;
      text-align: center;
      font: normal normal normal 16px "Lato, Bold";
      letter-spacing: 0px;
      background: var(--white) 0% 0% no-repeat padding-box;
      align-items: center;
      justify-content: center;
      color: var(--main-text);
      transition: all 0.1s;
      &.active {
        margin-left: 30px;
        /* background: var(--brand-blue) 0% 0% no-repeat padding-box; */
        color: var(--brand-blue);
        border-radius: 50px 0px 0px 50px;

        background: var(--brand-blue);
        background: -moz-linear-gradient(
          90deg,
          var(--brand-blue) 5%,
          var(--white) 5%
        );
        background: -webkit-linear-gradient(
          90deg,
          var(--brand-blue) 5%,
          var(--white) 5%
        );
        background: linear-gradient(
          90deg,
          var(--brand-blue) 5%,
          var(--white) 5%
        );
        filter: progid:DXImageTransform.Microsoft.gradient(startColorstr="#005ba7",endColorstr="#ffffff",GradientType=1);
      }
      &:hover:not(.active) {
        background: var(--faded-gray) 0% 0% no-repeat padding-box;
        color: var(--white);
      }
    }
  }
}

.darkmode-switch {
  justify-content: center;
  align-items: center;
  display: flex;
  -webkit-animation: fade-in 0.3s cubic-bezier(0.39, 0.575, 0.565, 1) both;
  animation: fade-in 0.3s cubic-bezier(0.39, 0.575, 0.565, 1) both;
}

.darkmode-switch__input {
  clip: rect(1px, 1px, 1px, 1px);
  clip-path: inset(50%);
  height: 1px;
  width: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
}

.darkmode-switch__label {
  position: relative;
  display: inline-block;
  width: 79px;
  height: 30px;
  background-color: #2b2b2b;
  border: 5px solid #5b5b5b;
  border-radius: 9999px;
  cursor: pointer;
  transition: all 0.4s cubic-bezier(0.46, 0.03, 0.52, 0.96);
}

.darkmode-switch__indicator {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%) translateX(-72%);
  display: block;
  width: 40px;
  height: 40px;
  background-color: #7b7b7b;
  border-radius: 9999px;
  box-shadow: 10px 0px 0 0 rgba(#000000, 0.2) inset;

  &::before,
  &::after {
    position: absolute;
    content: "";
    display: block;
    background-color: #ffffff;
    border-radius: 9999px;
  }

  &::before {
    top: 7px;
    left: 7px;
    width: 9px;
    height: 9px;
    background-color: #ffffff;
    opacity: 0.6;
  }

  &::after {
    bottom: 8px;
    right: 6px;
    width: 14px;
    height: 14px;
    background-color: #ffffff;
    opacity: 0.8;
  }
}

.darkmode-switch__decoration {
  position: absolute;
  top: 65%;
  left: 50%;
  display: block;
  width: 5px;
  height: 5px;
  background-color: #ffffff;
  border-radius: 9999px;
  animation: twinkle 0.8s infinite -0.6s;

  &::before,
  &::after {
    position: absolute;
    display: block;
    content: "";
    width: 5px;
    height: 5px;
    background-color: #ffffff;
    border-radius: 9999px;
  }

  &::before {
    top: -10px;
    left: 10px;
    opacity: 1;
    animation: twinkle 0.6s infinite;
  }

  &::after {
    top: -2px;
    left: 20px;
    animation: twinkle 0.6s infinite -0.2s;
  }
}

@keyframes twinkle {
  50% {
    opacity: 0.2;
  }
}

.darkmode-switch__indicator {
  &,
  &::before,
  &::after {
    transition: all 0.4s cubic-bezier(0.46, 0.03, 0.52, 0.96);
  }
}

.darkmode-switch__input:checked + .darkmode-switch__label {
  background-color: #8fb5f5;
  border-color: #347cf8;

  .darkmode-switch__indicator {
    background-color: #ecd21f;
    box-shadow: none;
    transform: translate(-50%, -50%) translateX(72%);

    &::before,
    &::after {
      display: none;
    }
  }

  .darkmode-switch__decoration {
    top: 50%;
    transform: translate(0%, -50%);
    animation: cloud 8s linear infinite;

    width: 20px;
    height: 20px;

    &::before {
      width: 10px;
      height: 10px;
      top: auto;
      bottom: 0;
      left: -8px;
      animation: none;
    }

    &::after {
      width: 15px;
      height: 15px;
      top: auto;
      bottom: 0;
      left: 16px;
      animation: none;
    }

    &,
    &::before,
    &::after {
      border-radius: 9999px 9999px 0 0;
    }

    &::after {
      border-bottom-right-radius: 9999px;
    }
  }
}

@keyframes cloud {
  0% {
    transform: translate(0%, -50%);
  }
  50% {
    transform: translate(-50%, -50%);
  }
  100% {
    transform: translate(0%, -50%);
  }
}
