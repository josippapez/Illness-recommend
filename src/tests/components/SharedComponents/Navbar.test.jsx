/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import Navbar from '../../../components/SharedComponents/Navbar/Navbar';

jest.mock('../../../store/actions');

const initialState = {
  theme: {
    darkTheme: false,
  },
  user: {
    id: 1,
    name: 'Anno Anno',
    email: 'test@tt.com',
    role: 'user',
    isAuthenticated: true,
    error: null,
    status: null,
    message: null,
  },
  patient: {
    currentPatientInfo: {
      id: 6,
      created: '2021-08-24T13:15:31.363Z',
      oib: '4657',
      name: 'test',
      age: 77,
      weight: 66,
      pregnantOrBreastFeed: true,
      symptomsSelected: [],
      medicationsSelected: [],
      alergies: [
        {
          id: 1,
          name: 'Amikacin',
        },
      ],
    },
    viewingPatientInfo: null,
    error: null,
    status: null,
    message: null,
    patientList: null,
  },
};

const mockStore = configureStore();
const store = mockStore(initialState);

describe('Navbar', () => {
  let component;

  const history = createMemoryHistory();

  it('renders correctly', () => {
    component = mount(
      <Provider store={store}>
        <Router history={history}>
          <Navbar />
        </Router>
      </Provider>
    );
    expect(component.html()).toMatchSnapshot();
  });
});
