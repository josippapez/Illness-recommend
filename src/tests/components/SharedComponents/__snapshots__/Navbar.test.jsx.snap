// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `"<div class=\\"navbar\\"><div class=\\"navbar-logo\\"><img class=\\"navbar-logo__image\\" src=\\"Logo.svg\\" alt=\\"logo-image\\"></div><div class=\\"darkmode-switch\\"><input type=\\"checkbox\\" class=\\"darkmode-switch__input\\" id=\\"Switch\\" readonly=\\"\\" checked=\\"\\"><label class=\\"darkmode-switch__label\\" for=\\"Switch\\"><span class=\\"darkmode-switch__indicator\\"></span><span class=\\"darkmode-switch__decoration\\"></span></label></div><div class=\\"navbar-links\\"><a aria-current=\\"page\\" class=\\"navbar-item active\\" href=\\"/\\"><img src=\\"home.svg\\" alt=\\"home\\" class=\\"navbar-icon\\">Početna stranica</a><div><a class=\\"navbar-item\\" href=\\"/patient-details\\"><img src=\\"resume.svg\\" alt=\\"home\\" class=\\"navbar-icon\\">Detalji pacijenta</a><a class=\\"navbar-item\\" href=\\"/medication-suggestion\\"><img src=\\"medicine.svg\\" alt=\\"home\\" class=\\"navbar-icon\\">Prijedlog lijekova</a></div><a class=\\"navbar-item\\" href=\\"/patients-details-history\\"><img src=\\"resume.svg\\" alt=\\"home\\" class=\\"navbar-icon\\">Prošli pacijenti</a><button class=\\"navbar-item\\"><img src=\\"login-arrow-symbol-entering-back-into-a-square.svg\\" alt=\\"home\\" class=\\"navbar-icon\\">Odjava</button></div></div>"`;
